lib/ngx/ssl/clienthello.lua

```diff
@@ -102,6 +107,39 @@ function _M.get_client_hello_server_name()
     return nil, ffi_str(errmsg[0])
 end
 
+-- return ext, err
+function _M.get_client_hello_ext_present()
+    local r = get_request()
+    if not r then
+        error("no request found")
+    end
+
+    if ngx_phase() ~= "ssl_client_hello" then
+        error("API disabled in the current context")
+    end
+
+    local sizep = get_size_ptr()
+    local intp = ffi.new("int*[1]") -- array of len 1 of int pointers
```

Comment:
```suggestion
    local intp = ffi.new("int*[1]") -- array of len 1 of int pointers
```
put the intp outside of this function as an upvalue